android/app/src/main/java/org/fossasia/openevent/adapters/FeedAdapter.java

```diff
@@ -0,0 +1,148 @@
+package org.fossasia.openevent.adapters;
+
+import android.app.Activity;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.support.design.widget.Snackbar;
+import android.support.graphics.drawable.VectorDrawableCompat;
+import android.support.v7.widget.RecyclerView;
+import android.text.Html;
+import android.text.TextUtils;
+import android.text.method.LinkMovementMethod;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import org.fossasia.openevent.OpenEventApp;
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.facebook.CommentItem;
+import org.fossasia.openevent.data.facebook.FeedItem;
+import org.fossasia.openevent.utils.DateUtils;
+import org.fossasia.openevent.utils.Utils;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import timber.log.Timber;
+
+/**
+ * Created by rohanagarwal94 on 08/6/17.
+ */
+public class FeedAdapter extends BaseRVAdapter<FeedItem, FeedAdapter.RecyclerViewHolder> {
+
+    private Activity activity;
+    private List<FeedItem> feedItems;
+    private AdapterCallback mAdapterCallback;
+    private ArrayList<CommentItem> commentItems;
+
+    class RecyclerViewHolder extends RecyclerView.ViewHolder {
+
+        @BindView(R.id.post_timestamp)
+        TextView timeStamp;
+        @BindView(R.id.txt_status_msg)
+        TextView statusMsg;
+        @BindView(R.id.txt_url)
+        TextView url;
+        @BindView(R.id.feed_image)
+        ImageView feedImageView;
+        @BindView(R.id.comment_button)
+        Button getComments;
+
+        RecyclerViewHolder(View view) {
+            super(view);
+            ButterKnife.bind(this, view);
+            view.bringToFront();
+
+            getComments.setOnClickListener(v -> {
+                FeedItem clickedFeedItem = feedItems.get(getPosition());
+                commentItems = new ArrayList<>();
+                if(clickedFeedItem.getComments() != null) {
+                    commentItems.addAll(clickedFeedItem.getComments().getData());
+                }
+                if(commentItems.size()!=0)
+                    mAdapterCallback.onMethodCallback(commentItems);
+                else
+                    Snackbar.make(v, activity.getResources().getString(R.string.no_comments), Snackbar.LENGTH_SHORT).show();
+            });
+        }
+    }
+
+    public FeedAdapter(Activity activity, List<FeedItem> feedItems) {
+        super(feedItems);
+        this.feedItems=feedItems;
+        this.activity = activity;
+        try {
+            this.mAdapterCallback = ((AdapterCallback) activity);
+        } catch (ClassCastException e) {
+            throw new ClassCastException("Activity must implement AdapterCallback.");
```

Comment:
activity is not the argument here. This is just a class cast. The interface is being implemented in the main activity to display a comments dialogfragment on comments button click.